### ISSUE TITLE
"TrainingData" and "StemmingNGrammProcessor" method

### ISSUE THREAD

**Issue by stenpiren** [digital-thinking/text-classification#2]
how to import "TrainingData" and "StemmingNGrammProcessor" methods? I am writing to run the example in README, and cannot figure out how to load them, any help?

**Comment by stenpiren** [comment]
> how to import "TrainingData" and "StemmingNGrammProcessor" methods? I am writing to run the example in README, and cannot figure out how to load them, any help?

I did some modifications like changing the location of TrainingData.java from ```integration-test``` folder to ```main``` folder. Also I noticed that the ```StemmingNgrammProcessor``` is nowhere defined, I changed it to ```DefaultProcessor```.

Looks like this:
`
package de.ixeption.classify;

import de.ixeption.classify.binary.BinaryUtils;
import de.ixeption.classify.binary.svm.BinaryTextClassifierTrainer;
import de.ixeption.classify.features.FeatureUtils;
import de.ixeption.classify.features.TextFeature;
import de.ixeption.classify.features.impl.BagOfWordsFeatureExtractor;
import de.ixeption.classify.pipeline.impl.DefaultTextPipeline;
import de.ixeption.classify.postprocessing.impl.DefaultProcessor;
import de.ixeption.classify.tokenization.impl.NormalizingTextTokenizer;
import de.ixeption.classify.TrainingData;
import jdk.nashorn.internal.parser.Token;
import java.util.stream.Stream;

import java.util.ArrayList;
import java.util.Arrays.stream;
import java.util.Set;
import java.util.stream.Collectors;

public class demo {
    public static void main(String[] args) {
        // https://www.kaggle.com/c/si650winter11/data
        TrainingData data = new TrainingData("data/training.csv");
        ArrayList<TextFeature> textFeatures = data.getTextFeatures();
        ArrayList<Integer> labels = data.getLabels();
        FeatureUtils.shuffle(textFeatures, labels);

        NormalizingTextTokenizer normalizingTextTokenizer = new NormalizingTextTokenizer();
        DefaultProcessor filteringProcessor = new DefaultProcessor(3, 3, 25);

        Set<String> corpus = textFeatures.stream().map(normalizingTextTokenizer::tokenize).map(filteringProcessor::process).flatMap(Arrays::stream).collect(Collectors.toSet());
        DefaultTextPipeline pipeline = new DefaultTextPipeline(new BagOfWordsFeatureExtractor(corpus), normalizingTextTokenizer);
        pipeline.getTokenProcessors().add(filteringProcessor);

        BinaryTextClassifierTrainer binaryTextClassifierTrainer = new BinaryTextClassifierTrainer(0.5, 0.5, pipeline);

        // cross validate
        BinaryUtils.BinaryConfusionMatrixMeasure confusionMatrixMeasure = binaryTextClassifierTrainer
                .crossValidate(textFeatures.toArray(new TextFeature[0]), labels.stream().mapToInt(Integer::intValue).toArray());
        //log.info(" confusion matrix\n" + confusionMatrixMeasure);

        // train and validate
        int trainSize = 6000;
        TextFeature[] trainX = textFeatures.stream().limit(trainSize).toArray(TextFeature[]::new);
        int[] trainY = labels.stream().limit(trainSize).mapToInt(Integer::intValue).toArray();

        TextFeature[] testX = textFeatures.stream().skip(trainSize).toArray(TextFeature[]::new);
        int[] testY = labels.stream().skip(trainSize).mapToInt(Integer::intValue).toArray();

        TextClassifier classifier = binaryTextClassifierTrainer.train(trainX, trainY, null);
        int[] predictions = Arrays.stream(testX).map(classifier::predict).mapToInt(Prediction::getLabel).toArray();

        int correct = 0;
        for (int i = 0; i < testY.length; i++) {
            if (predictions[i] == testY[i]) {
                correct++;
            }

        }

        double accurancy = correct / (double) testY.length;
        //log.info("Accuracy: " + accurancy);
    }
}

`
However, I am still unable to run as the ```Arrays::stream``` throws an error: "cannot resolve method stream" also, the "log" which library is it?

**Comment by digital-thinking** [comment]
Hi @stenpiren,

which Java version do you use? Arrays::stream should be there since Java 8.  It might not work for Android Java, as it does not support all Java 8 features.

**Comment by stenpiren** [comment]
> Hi @stenpiren,
> 
> which Java version do you use? Arrays::stream should be there since Java 8. It might not work for Android Java, as it does not support all Java 8 features. See [here](https://developer.android.com/studio/write/java8-support#supported-features).

I see. I am using Java 1.8, I guess is Java 8 (I am new to this). Can you tell also, which library does "Token" and "log" get imported from the the example from README? Thanks.

**Comment by digital-thinking** [comment]
Should be visible [here](https://github.com/digital-thinking/text-classification/blob/master/src/integration-test/java/de/ixeption/classify/TFIDFBagOfWordsIT.java). You can simply clone the repository and execute the test, should work, if you have the correct Java version and maven installed.
